sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -622,7 +628,114 @@ trait GpuHashJoin extends GpuExec {
     (joinLeft.output.size, boundCondition)
   }
 
-  def doJoin(
+  private def existenceJoinIterator(
+    builtBatch: ColumnarBatch,
+    stream: Iterator[ColumnarBatch]) = new Iterator[ColumnarBatch]()
+      with AutoCloseable with Arm {
+    var closed: Boolean = false
+
+    // iteration-independent resources
+    val resources = ArrayBuffer[AutoCloseable]()
+    def use[T <: AutoCloseable](ac: T): T = {
+      resources += ac
+      ac
+    }
+
+    val compiledConditionRes: Option[(Table, CompiledExpression)] = boundCondition.map(gpuExpr => (
+        use(GpuColumnVector.from(builtBatch)),
+        use(gpuExpr.convertToAst(numFirstConditionTableColumns).compile())
+      )
+    )
+
+    val rightKeysTab = use(
+      withResource(GpuProjectExec.project(builtBatch, boundBuildKeys))(GpuColumnVector.from(_)))
+
+    val falseScalar = use(Scalar.fromBool(false))
+    val trueScalar = use(Scalar.fromBool(true))
+    override def hasNext: Boolean = {
+      val streamHasNext = stream.hasNext
+      if (!streamHasNext) {
+        close()
+      }
+      streamHasNext
+    }
+
+    override def next(): ColumnarBatch = {
+      try {
+        withResource(stream.next()) { leftColumnarBatch =>
+          existenceJoinNextBatch(leftColumnarBatch)
+        }
+      } catch {
+        case t: Throwable =>
+          close()
+          throw t
+      }
+    }
+
+    override def close() = if (!closed) {
+      closed = true
+      val resourcesReversed = resources.reverse
+      resourcesReversed.safeClose()
+    }
+
+    private def leftKeysTable(leftColumnarBatch: ColumnarBatch): Table = {
+      withResource(GpuProjectExec.project(leftColumnarBatch, boundStreamKeys))(
+        leftKeys => GpuColumnVector.from(leftKeys))
+    }
+
+    private def conditionalBatchLeftSemiJoin(
+      leftColumnarBatch: ColumnarBatch,
+      rightTab: Table,
+      leftfKeysTab: Table,
+      compiledCondition: CompiledExpression): GatherMap = {
+      withResource(GpuColumnVector.from(leftColumnarBatch))(leftTab =>
+        Table.mixedLeftSemiJoinGatherMap(
+          leftfKeysTab,
+          rightKeysTab,
+          leftTab,
+          rightTab,
+          compiledCondition,
+          if (compareNullsEqual) NullEquality.EQUAL else NullEquality.UNEQUAL))
+    }
+
+    private def existsScatterMap(leftColumnarBatch: ColumnarBatch): GatherMap = {
```

Comment:
Would be good to comment this function or have a comment somewhere that documents how the existence join column is computed.